libs/remix-astwalker/src/astWalker.ts

```diff
@@ -58,62 +58,131 @@ export class AstWalker extends EventEmitter {
       }
     }
   }
+
+  normalizeNodes(nodes: AstNode[]): AstNode[] {
+    // Remove null, undefined and empty elements if any
+    nodes = nodes.filter(e => e)
+
+    // If any element in nodes is array, extract its members
+    const objNodes = []
+    nodes.forEach(x => { 
+      if (Array.isArray(x)) objNodes.push(...x)
+      else objNodes.push(x)
+    });
+    
+    // Filter duplicate nodes using id field
+    const normalizedNodes = []
+    objNodes.forEach((element) => {
+      const firstIndex = normalizedNodes.findIndex(e => e.id === element.id)
+      if(firstIndex == -1) normalizedNodes.push(element)
+    })
+    return normalizedNodes
+  }
+
+  getASTNodeChildren(ast: AstNode): AstNode[] {
+    const nodes = ast.nodes           // for ContractDefinition
```

Comment:
What would be the best way to share this with the solidity team in order to keep this up to date? or is this stable enough so we don't really need to be worried about potential changes.

Reply:
I will think about it.